agent-install/agent-install.sh

```diff
@@ -320,21 +392,20 @@ function get_all_variables() {
     get_variable AGENT_DEPLOY_TYPE 'device'
     get_variable AGENT_WAIT_MAX_SECONDS '30'
 
-    if [[ "${AGENT_DEPLOY_TYPE}" == "device" ]]; then
+    if is_device; then
         get_variable PKG_PATH '.'
         get_variable NODE_ID_MAPPING_FILE 'node-id-mapping.csv'
         get_variable PKG_APT_KEY
         get_variable APT_REPO_BRANCH 'updates'
-    elif [[ "${AGENT_DEPLOY_TYPE}" == "cluster" ]]; then
+    elif is_cluster; then
         get_variable EDGE_CLUSTER_STORAGE_CLASS 'gp2'
         get_variable AGENT_NAMESPACE 'openhorizon-agent'
         get_variable USE_EDGE_CLUSTER_REGISTRY 'true'
         get_variable AGENT_DEPLOYMENT_STATUS_TIMEOUT_SECONDS '75'
 
         if [[ "$USE_EDGE_CLUSTER_REGISTRY" == "true" ]]; then
-            #lily: do we really need the user to input IMAGE_ON_EDGE_CLUSTER_REGISTRY and AGENT_IMAGE_TAG. These seem like internal values (our code creates the image in the local registry with these values and then uses that image). Can we just always default them based on the edge cluster type?
+            #lily: can we default IMAGE_ON_EDGE_CLUSTER_REGISTRY by querying the edge cluster, so the user doesn't have to input it unless they want to override it?
```

Comment:
It is not hard for k3s and microk8s. But need to find a good way to get/parse the internal image registry host for OCP.